src/js/components/JsonViewer.js

```diff
@@ -5,8 +5,11 @@ import ArrayGroup from './ArrayGroup';
 export default class extends React.PureComponent {
     render = () => {
         const { props } = this;
-        const namespace = [props.name];
+        let namespace = [props.name];
         let ObjectComponent = JsonObject;
+        if (typeof props.name === 'object' && !Array.isArray(props.name)) {
+            namespace = ['ReactElement'];
```

Comment:
what if `namespace = props.displayName` ? 🤔

Reply:
Great idea! Added it.

Also handled If component is unnamed It will use `Anonymous` exactly the same as most of the react debug tools